lib/ansible/cli/doc.py

```diff
@@ -495,7 +495,9 @@ def display_plugin_list(self, results):
                     desc = desc[:linelimit] + '...'
 
                 pbreak = plugin.split('.')
-                if pbreak[-1].startswith('_'):  # Handle deprecated # TODO: add mark for deprecated collection plugins
+                # TODO: add mark for deprecated collection plugins
+                if pbreak[-1].startswith('_') and plugin.startswith('ansible.builtin.'):
```

Comment:
legacy also

Reply:
Right! Fixed in 7e65e8ced5c8434901d6595207d64c97dc549571.